pkg/operator/ceph/cluster/telemetry/telemetry.go

```diff
@@ -29,16 +29,40 @@ import (
 var logger = capnslog.NewPackageLogger("github.com/rook/rook", "telemetry")
 
 const (
-	RookVersionKey = `rook/version`
+	RookVersionKey             = "rook/version"
+	K8sVersionKey              = "rook/kubernetes/version"
+	CSIVersionKey              = "rook/csi/version"
+	MonMaxIDKey                = "rook/cluster/mon/max-id"
+	MonCountKey                = "rook/cluster/mon/count"
+	MonAllowMultiplePerNodeKey = "rook/cluster/mon/allow-multiple-per-node"
+	MonPVCEnabledKey           = "rook/cluster/mon/pvc/enabled"
+	MonStretchEnabledKey       = "rook/cluster/mon/stretch/enabled"
+	DeviceSetTotalKey          = "rook/cluster/storage/device-set/count/total"
+	DeviceSetPortableKey       = "rook/cluster/storage/device-set/count/portable"
+	DeviceSetNonPortableKey    = "rook/cluster/storage/device-set/count/non-portable"
+	NetworkProviderKey         = "rook/cluster/network/provider"
+	ExternalModeEnabledKey     = "rook/cluster/external-mode"
 )
 
-// SetRookVersion sets the Rook version in Ceph's config-key store to allow Rook clusters that
+var CSIVersion string
+
+// ReportRookVersion sets the Rook version in Ceph's config-key store to allow Rook clusters that
 // enable Ceph telemetry to be identified easily as Rook clusters while additionally providing
 // useful information about the version of Rook that is managing the cluster.
 // e.g., rook/version=v1.8.7
-func SetRookVersion(context *clusterd.Context, clusterInfo *client.ClusterInfo) {
+func ReportRookVersion(context *clusterd.Context, clusterInfo *client.ClusterInfo) {
 	ms := config.GetMonStore(context, clusterInfo)
 	if err := ms.SetKeyValue(RookVersionKey, rookversion.Version); err != nil {
-		logger.Warningf("failed to set telemetry key; this cluster may not be identifiable by ceph telemetry as a Rook cluster. %v", err)
+		logger.Warningf("failed to set rook version telemetry key; this cluster may not be identifiable by ceph telemetry as a Rook cluster. %v", err)
+	}
+	logger.Debugf("set telemetry key: %s=%s", RookVersionKey, rookversion.Version)
+}
+
+func ReportKeyValue(context *clusterd.Context, clusterInfo *client.ClusterInfo, key, value string) {
```

Comment:
```suggestion
func ReportKeyValue(context *clusterd.Context, clusterInfo *client.ClusterInfo, key, value string) {
```
function `ReportKeyValue` seems similar to `ReportRookVersion` should we reuse the same method?

Reply:
There is a different warning message when it fails since the rook version key is more critical to the collection, so I left it as a separate method. @BlaineEXE wdyt?

Reply:
I think it feels okay either way.